Fix dashboard case & death count

gqcorneby opened the pull request:
### :pushpin: References

-   **Issue:** Closes #86998zrpc  [Defect - Zebra Dashboard Case & Death Count](https://app.clickup.com/t/86998zrpc)

### :memo: Implementation
- fetch only case program for cases and deaths count


### :video_camera: Screenshots/Screen capture
- Only one with small a small discrepancy is COVID and this could be due to cases without a dataSource which is counted in the dashboard but not in the event tracker page

![image](https://github.com/user-attachments/assets/89877566-8801-4fd8-8524-8519b8a26eda)

COVID
dashboard - 16484
event tracker - 16338 + 142 = 16480 - there may be 4 that do not have a dataSource
<img width="400px" src="https://github.com/user-attachments/assets/e4955104-be21-4dff-8265-4fa1727db5aa" /> <img width="400px" src="https://github.com/user-attachments/assets/95e9bda1-a147-4eb9-a286-325b640a9e18" />

ANTHRAX
<img width="400px" src="https://github.com/user-attachments/assets/607ff568-8082-4c0f-98db-51d16eb0135a"/> <img width="400px" src="https://github.com/user-attachments/assets/9e29c6de-ced8-4e36-ac1e-d1ca8e03209d"/>

BACTERIAL MENINGITIS
<img width="400px" src="https://github.com/user-attachments/assets/f8d8de99-ae1b-44c3-83f9-7f535cda206f"/> <img width="400px" src="https://github.com/user-attachments/assets/8db63e5e-ad8d-4d22-9d86-e97de38da9c8"/>

PLAGUE
<img width="400px" src="https://github.com/user-attachments/assets/e6a7f856-a2bc-4b21-87b9-0d07c33da999"/> <img width="400px" src="https://github.com/user-attachments/assets/0b14cc78-ca87-40ae-87a0-57d03cc8c417"/>

MEASLES 
<img width="400px" src="https://github.com/user-attachments/assets/2dd994e7-655f-4808-86d7-b3662259f570"/> <img width="400px" src="https://github.com/user-attachments/assets/a437bd7e-fd8d-441a-9c84-78814b217682"/>


### :fire: Notes to the tester

gqcorneby committed:
use cases for deaths and cases in national dashboard

anagperal reviewed (approved):
Thanks @gqcorneby !! All ok, but only one thing if you can remove ALERTS_PROGRAM_EVENT_TRACKER_OVERVIEW_DATASTORE_KEY that is not used?

gqcorneby committed:
clean up unused reference

gqcorneby committed:
fix: remove caseDataSource filter in building the cases and deaths count

gqcorneby committed:
Merge branch 'development' into feature/national-dashboard-cases-deaths

gqcorneby committed:
fix: national dashboard overview cases and deaths
- use all if user defined. Otherwise, ND1 or ND2

gqcorneby committed:
fix: default dataSource in event form
- undefined if user defined

gqcorneby commented:
@anagperal requesting review again. The displayed cases and deaths in the performance overview should be dependent on the event data source. Following:
1. if case data source is user defined: show total cases and deaths
2. if eIDSR: show either ND1 (default) or ND2.

Additional fixes:
- Also updated the default dataSource in the event form. Data source should be undefined if case data source is user defined
- Event page scrolls down to the charts when iframe loads. After investigating, this is because there's a field inside the iframe. Inside `#dhis2-portal-root`,  `<input placeholder=​"Filter by name" type=​"search" class=​"jsx-3353877153 jsx-31445346 dense" value>​` has some autofocus behavior which forces the scroll. I updated the styles to use `display:none` instead of `visibility:hidden` when loading which allowed this behavior to be ignored. 


https://github.com/user-attachments/assets/c603bbd4-dfd4-4671-9915-6adee6fa8978

gqcorneby committed:
fix: scrolling upon loading

anagperal reviewed (approved):
thanks @gqcorneby !!  All works! it's really great not scroll the event tracker page :)
